interface/wx/webrequest.h

```diff
@@ -0,0 +1,611 @@
+    @category{net}
+
+    @see wxWebRequest
+*/
+class wxWebResponse
+{
+public:
+    /**
+        Returns the final URL.
+        This URL might be different than the request URL when a redirection
+        occurred.
+    */
+    wxString GetURL() const;
+
+    /**
+        Returns a header from the response or an empty string if the header
+        could not be found.
+
+        @param name Name of the header field
+    */
+    wxString GetHeader(const wxString& name) const;
+
+    /**
+        Returns the MIME type of the response (if available).
+    */
+    wxString GetMimeType() const;
+
+    /**
+        Returns the status code returned by the server.
+    */
+    int GetStatus() const;
+
+    /**
+        Returns the status text of the response.
+    */
+    wxString GetStatusText() const;
+
+    /**
+        Returns a stream which represents the response data sent by the server.
+    */
+    wxInputStream* GetStream();
+
+    /**
+        Returns a suggested filename for the response data.
+    */
+    wxString GetSuggestedFileName() const;
+
+    /**
+        Returns all response data as a string.
+
+        @param conv wxMBConv used to convert the response to a string.
+                    If @c NULL, the conversion will be determined by
+                    response headers. The default is UTF-8.
+    */
+    wxString AsString(wxMBConv* conv = NULL) const;
+};
+
+/**
+    @class wxWebSession
+
+    This class handles session-wide parameters and data used by wxWebRequest
+    instances.
+
+    Usually the default session available via wxWebSession::GetDefault()
+    should be used. Additional instances might be useful if session separation
+    is required. Instances <b>must not</b> be deleted before every active web
+    request has finished.
+
+    Every wxWebRequest sharing the same session object will use the same
+    cookies. Additionally, an underlying network connection might be kept
+    alive to achieve faster additional responses.
+
+    @since 3.1.2
+
+    @library{wxnet}
+    @category{net}
+
+    @see wxWebRequest
+*/
+class wxWebSession
+{
+public:
+    /**
+        Constructor for the session
+
+        All requests created by a call to CreateRequest() will use this session
+        for communication and to store cookies.
+    */
+    wxWebSession();
+
+    /**
+        Create a new request for the specified URL
+
+        @param url
+            The URL of the HTTP resource for this request
+        @param id
+            Optional id sent with events
+    */
+    wxWebRequest* CreateRequest(const wxString& url, int id = wxID_ANY);
+
+    /**
+        Sets a request header in every wxWebRequest created from this session
+        after is has been set.
+
+        A good example for a session-wide request header is the @c User-Agent
+        header.
+
+        @param name Name of the header
+        @param value String value of the header
+    */
+    void SetHeader(const wxString& name, const wxString& value);
+
```

Comment:
Add documentation of
```
const wxWebRequestHeaderMap& GetHeaders() const;
```

Reply:
I've removed this one from public API for now.